src/fourier.jl

```diff
@@ -0,0 +1,771 @@
+function positive_fft_bins(n_bin::Integer; include_zero::Bool = false)
+    minbin = 2
+    if include_zero
+        minbin = 1
+    end
+    return (minbin : (n_bin+1) ÷ 2)
+end
+
+function poisson_level(norm::String; meanrate = nothing, n_ph = nothing, backrate::Real = 0.0)
+    if norm == "abs"
+        return 2.0 * meanrate
+    elseif norm == "frac"
+        return 2.0 / (meanrate - backrate)^2 * meanrate
+    elseif norm == "leahy"
+        return 2.0
+    elseif norm == "none"
+        return float(n_ph)
+    else 
+        throw(ArgumentError("Unknown value for norm: $norm"))
+    end
+end
```

Comment:
I think a better way to represent this is to define a "normalisation" type.  Something along these lines (also with better names than `Abs` and `Frac`, I'm just copying the short names used here):
```julia
abstract type Normalization end

struct Abs{T} <: Normalization
    meanrate::T
end
poisson_level(n::Abs) = 2 * n.meanrate

struct Frac{T} <: Normalization
    meanrate::T
    backrate::T
end
poisson_level(n::Frac) = 2 / (n.meanrate - n.backrate) ^ 2 * n.meanrate

struct Leahy <: Normalization end
poisson_level(::Leahy) = 2

struct Unnormalized{T} <: Normalization
    n_ph::T
end
poisson_level(n::Unnormalized) = n_ph
```
It's idiomatic, it makes natural use of multiple dispatch (instead of dispatching on strings....), it's optimisable (the current implementation of `poisson_level` can't possibly be optimised by the compiler, since all arguments are basically `Any`), no need to abuse `nothing`, etc.

Similar things can be done elsewhere throughout the code, to make it more idiomatic and actually use Julia features, instead of writing Python code with Julia syntax.  But let's not focus on this now, this is mainly an idea for the future, to give a concrete suggestion for how to make the code more idiomatic.

Reply:
It removes use of nothing only when there is no `isnothing` check for the variable. Giving a default value with one of the other arguments of the function is one of the things heavily used in the fourier.py and it would have to be changed completely to remove the `nothing`s where this check is happening. Also, wouldn't it be more user convenient to pass the type as argument rather than making user create objects for the dispatch (Like how `fit(Histogram, args...)` works in StatsBase) 
```julia
julia> struct Abs end

julia> poisson_level(::Type{Abs}, meanrate::Real) = 2.0 * meanrate
poisson_level (generic function with 1 method)

julia> poisson_level(Abs,9)
18.0
```
And yeah, I understand that many sections can be worked upon to be more idiomatic, and I will keep that in mind writing future codes.